src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilderExtensions.cs

```diff
@@ -61,7 +61,12 @@ public static ClusterBuilder ConfigureWithConnectionString(this ClusterBuilder b
             Ensure.IsNotNull(builder, nameof(builder));
             Ensure.IsNotNull(connectionString, nameof(connectionString));
 
-            connectionString = connectionString.Resolve();
+            if (connectionString.Scheme == ConnectionStringScheme.MongoDBPlusSrv)
+            {
+                var connectionMode = connectionString.Connect;
+                var resolveHosts = connectionMode == ClusterConnectionMode.Direct || connectionMode == ClusterConnectionMode.Standalone;
+                connectionString = connectionString.Resolve(resolveHosts);
```

Comment:
No longer resolve hosts this early when using SRV with anything except direcct or standalone.